whipper/common/program.py

```diff
@@ -470,6 +471,70 @@ def getHTOA(self):
         stop = track.getIndex(1).absolute - 1
         return start, stop
 
+    def getCoverArt(self, path, cover_art, release_id):
+        """
+        Get cover art image from Cover Art Archive if user chooses to
+
+        :type  cover_art: bool
+        :param cover_art: whether to fetch cover art or not
+        :type  path: str
+        :param path: where to store the fetched image
+        :type  release_id: str
+        :param release_id: a release id (self.program.metadata.mbid)
+
+        :rtype: str or None
+        """
+        if not cover_art:
+            return
+
+        import requests
+        import whipper
+
+        image_path = os.path.join(path, 'coverart.jpg')
+        user_agent = ("whipper/%s "
+                      "( https://github.com/whipper-team/whipper )" %
+                      whipper.__version__)
+        headers = {
+            'User-Agent': user_agent
+        }
+        url = 'https://coverartarchive.org/release/%r/front' % release_id
+
+        logger.debug('fetching cover art at: %r', url)
+        try:
+            resp = requests.get(url, stream=True, headers=headers)
+        except requests.exceptions.ConnectionError as e:
+            logger.error('error fetching cover art: %r', e)
+            return
+
+        if resp:
+            if resp.status_code == 200:
+                content_type = resp.headers['content-type']
+                allowed_img_fmts = {'image/jpeg', 'image/png'}
```

Comment:
We could also use `-500` or `-1200` "thumbnail" images to ensure always getting a JPEG. `-1200` should be getting a version large enough for any reasonable embedding and may indeed still be preferable to make smaller before embedding, but can also be missing from some older uploads (see [CAA-88](https://tickets.metabrainz.org/browse/CAA-88)). `-500` is available for all CAA images, but might maybe be too small for some people’s liking?

(The "500" and "1200" refer to the max dimension. So e.g., a 300×2000px² image will be scaled down to 75×500px² and 180×1200px² respectively. A 2000×2000px² image will be scaled down to 500×500px² and 1200×1200px² image respectively.)

Reply:
Well, I expect the majority of the cover art images to be roughly squared, am I wrong?

Reply:
Anyway I think that's a good starting point and should be enough for this pull request. We can always tweak this later...